docs/ops/image/Interpolate_4.md

```diff
@@ -377,54 +409,48 @@ class InterpolateCalculation:
-            in_x1[x] = max(0, min(int(in_x), input_width - 1))
-            in_x2[x] = min(in_x1[x] + 1, input_width - 1)
-            dx1[x] = abs(in_x - in_x1[x])
-            dx2[x] = abs(in_x - in_x2[x])
+            in_x = max(0.0, min(in_x, input_width - 1));
+
+            in_x1[x] = min(in_x, input_width - 1);
+            in_x2[x] = min(in_x1[x] + 1, input_width - 1);
 
-            if in_x1 == in_x2:
+            dx1[x] = abs(in_x - in_x1[x]);
+            dx2[x] = abs(in_x - in_x2[x]);
+            if in_x1[x] == in_x2[x]:
                 dx1[x] = 0.5
                 dx2[x] = 0.5
 
-        for n in range(batch_size):
-            for c in range(num_channels):
-                for y in range(output_height):
-                    for x in range(output_width):
+        for n in range(0, batch_size):
+            for c in range(0, num_channels):
+                for y in range(0, output_height):
+                    for x in range(0, output_width):
                         x11 = reshaped_data[n, c, in_y1[y], in_x1[x]]
                         x21 = reshaped_data[n, c, in_y1[y], in_x2[x]]
                         x12 = reshaped_data[n, c, in_y2[y], in_x1[x]]
                         x22 = reshaped_data[n, c, in_y2[y], in_x2[x]]
-                        temp = dx2[x] * dy2[y] * x11 + dx1[x] * dy2[y] * x21
-                        temp += dx2[x] * dy1[y] * x12 + dx1[x] * dy1[y] * x22
+                        temp = dx2[x] * dy2[y] * x11 + dx1[x] * dy2[y] * x21 + dx2[x] * dy1[y] * x12 + dx1[x] * dy1[y] * x22
                         result[n, c, y, x] = temp
 
         return np.reshape(result, self.output_shape)
 
     def nearest_interpolation(self, input_data):
-        if not ('nearest_mode' in self.attrs):
-            self.attrs['nearest_mode'] = 'floor'
-
-        self.get_nearest_pixel = GetNearestPixel(attrs['nearest_mode'])
-
         result = np.zeros(self.output_shape)
 
         num_of_axes = len(self.axes)
-        for coordinates in np.ndindex(self.output_shape):
+        for coordinates in np.ndindex(tuple(self.output_shape)):
             input_coords = np.array(coordinates, dtype=np.int64)
-            for i in range(num_of_axes):
-                axis = self.axes[i]
-                in_coord = self.get_original_coordinate(coordinates[axis], self.scales[axis], self.output_shape[axis], self.input_shape[axis])
-                nearest_pixel = self.get_nearest_pixel(in_coord, self.scales[axis] < 1)
+            for i, axis in enumerate(self.axes):
+                in_coord = self.get_original_coordinate(coordinates[axis], self.scales[i], self.output_shape[axis], self.input_shape[axis])
+                nearest_pixel = self.get_nearest_pixel(in_coord, self.scales[i] < 1)
                 input_coords[axis] = max(0, min(nearest_pixel, self.input_shape[axis] - 1))
-           result[coordinates] = input_data[input_coords]
+            result[coordinates] = input_data[tuple(input_coords)]
 
         return result
 ```
```

Comment:
In below Example: axes is not attribute anymore, align_corners is one type in coordinate_transformation_mode attribute.

Reply:
Fixed.